### PR TITLE
[New Extension] Image Modification

### PR DESCRIPTION
**Pull request by SKaplanOfficial**
## Description

This extension provides easy access to basic image modification operations such as flipping, rotating, resizing, scaling, and converting between formats. The extension uses the built-in `sips` Terminal command to modify selected image files.

## Commands

- Convert Images
    - Convert selected images from their current format to another.
- Flip Images Horizontally
    - Flip selected images horizontally.
- Flip Images Vertically
    - Flip selected images vertically.
- Resize Images [Width] [Height]
    - Resize images proportionally by specifying either width or height.
    - Resize images precisely by specifying both parameters.
- Rotate Images [Degrees]
    - Rotate images clockwise by the specified amount.
- Scale Images [Scale Factor]
    - Scale images proportionally by the specified factor.

## Screencast

[SIPS720.webm](https://user-images.githubusercontent.com/7865925/220980602-91a5021a-d5ba-4c32-a5a3-3e2602a2be70.webm)

[YouTube Video](https://www.youtube.com/watch?v=nGbdiaJNEsc)

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/sips/src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            copy (POSIX path of (item i of theSelection as alias)) to end of thePaths
+          end if
+        end repeat
+        return thePaths
+      end if
+    end tell`
+  );
+
+  return selectedImages.split(", ");
+};
```
Comment (pernielsentikaer) [inline extensions/sips/src/utils.ts:26]: You can get this from Raycast with [getSelectedFinderItems](https://developers.raycast.com/api-reference/environment#getselectedfinderitems)

File: extensions/sips/src/scale.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        const oldWidth = parseInt(resultArr[4]);
+        const oldHeight = parseInt(resultArr[8]);
+
+        execSync(`sips --resampleHeightWidth ${oldHeight * scaleNumber} ${oldWidth * scaleNumber} "${imagePath}"`);
+      }
+
+      await showToast({ title: `Scaled ${selectedImages.length.toString()} ${pluralized}` });
+    } catch (error) {
+      console.log(error);
+      await showToast({ title: `Failed to scale ${selectedImages.length.toString()} ${pluralized}` });
```
Comment (pernielsentikaer) [inline extensions/sips/src/scale.ts:37]: Missing `style: Toast.Style.Failure`

File: extensions/sips/src/rotate.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  }
+
+  if (selectedImages) {
+    const pluralized = `image${selectedImages.length === 1 ? "" : "s"}`;
+    try {
+      const pathStrings = '"' + selectedImages.join('" "') + '"';
+      execSync(`sips --rotate ${degrees} ${pathStrings}`);
+      await showToast({ title: `Rotated ${selectedImages.length.toString()} ${pluralized} by ${degrees} degrees` });
+    } catch {
+      await showToast({ title: `Failed to rotate ${selectedImages.length.toString()} ${pluralized}` });
```
Comment (pernielsentikaer) [inline extensions/sips/src/rotate.ts:21]: Missing `style: Toast.Style.Failure`

File: extensions/sips/src/flip-vertically.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  }
+
+  if (selectedImages) {
+    const pluralized = `image${selectedImages.length === 1 ? "" : "s"}`;
+    try {
+      const pathStrings = '"' + selectedImages.join('" "') + '"';
+      execSync(`sips --flip vertical ${pathStrings}`);
+      await showToast({ title: `Flipped ${selectedImages.length.toString()} ${pluralized} vertically` });
+    } catch {
+      await showToast({ title: `Failed to flip ${selectedImages.length.toString()} ${pluralized}` });
```
Comment (pernielsentikaer) [inline extensions/sips/src/flip-vertically.ts:20]: Missing `style: Toast.Style.Failure`

File: extensions/sips/src/flip-horizontally.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  }
+
+  if (selectedImages) {
+    const pluralized = `image${selectedImages.length === 1 ? "" : "s"}`;
+    try {
+      const pathStrings = '"' + selectedImages.join('" "') + '"';
+      execSync(`sips --flip horizontal ${pathStrings}`);
+      await showToast({ title: `Flipped ${selectedImages.length.toString()} ${pluralized} horizontally` });
+    } catch {
+      await showToast({ title: `Failed to flip ${selectedImages.length.toString()} ${pluralized}` });
```
Comment (pernielsentikaer) [inline extensions/sips/src/flip-horizontally.ts:20]: Missing `style: Toast.Style.Failure`

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you 😊

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/sips/src/utils.ts (hunk 1 above)
Comment (SKaplanOfficial) [reply extensions/sips/src/utils.ts:26]: I'd like to use the items' data kind, which to my understanding is not necessarily the same as the file extension, to filter the list of selected items. I don't think that is possible with Raycast's `getSelectedFinderItems` -- correct me if I'm wrong though.

Comment (pernielsentikaer) [reply extensions/sips/src/utils.ts:26]: No that's correct, you can use this if you want 😊

let me know 😊

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋

Looks good to me, approved 🔥

**Pull request merged**